pennylane/vqe/vqe.py

```diff
@@ -58,6 +54,15 @@ class Hamiltonian:
       (-0.543) [Z0 H2]
     + (0.2) [X0 Z1]
 
+    The coefficients can be a trainable tensor, for example:
+
+    >>> coeffs = tf.Variable([0.2, -0.543], dtype=tf.double)
+    >>> obs = [qml.PauliX(0) @ qml.PauliZ(1), qml.PauliZ(0) @ qml.Hadamard(2)]
+    >>> H = qml.Hamiltonian(coeffs, obs)
+    >>> print(H)
+      (-0.543) [Z0 H2]
+    + (0.2) [X0 Z1]
```

Comment:
Question: should the print statement use tf.Variables as coefficients instead? Could be ugly, but contains more info...

Reply:
I think so yes. It would be nice if `H.__str__` was ambivalent to the coefficient type, and simply called the coeff object's `__str__`/`__repr__`.

Reply:
Interesting, so it does this via `f"{coeff}"`, but this calls the tensor's ``__format__()`` function which returns a representation of the value without any tensor info! 

I tried to use `f"{coeff.__str__()}"`, but it does look a bit clumsy:
![Screenshot from 2021-08-09 10-33-48](https://user-images.githubusercontent.com/23052011/128679153-ef765162-81cc-4bf1-93e0-5e4098075cb5.png)

Any preference?